x-pack/legacy/plugins/ingest/server/libs/adapters/policy/adapter_types.ts

```diff
@@ -6,106 +6,193 @@
-    inputs: t.array(t.object),
-  },
-  'NewDatasource'
-);
+/**
+ * A logical grouping of places where data is coming from, such as "Production", "Staging",
+ * "Production East-1", "Metrics Cluster", etc. -- these groupings are user-defined. We
+ * store information collected from the user about this logical grouping such as a name and
+ * any other information we need about it to generate the associated config. A package
+ * defines its own data source templates that can use user-provided values to generate the
+ * data source config. A single data source will typically enable users to collect both logs
+ * and metrics. A data source can be in multiple policies at the same time. A datasource can
+ * have multiple streams.
+ */
+export interface Datasource {
+  id: string;
+  name: string;
+  package: Package;
+  read_alias?: string;
+  streams: Stream[];
+}
 
-export const NewRuntimePolicyFile = t.interface(
-  {
-    name: t.string,
-    description: t.string,
-    monitoring_enabled: t.boolean,
-    shared_id: t.string,
-    version: t.number,
-    status: t.keyof({
-      active: null,
-      locked: null,
-      inactive: null,
-    }),
-    agent_version: t.string,
-    data_sources: t.array(RuntimeDatasource),
-    created_on: t.string,
-    created_by: t.union([t.undefined, t.string]),
-    updated_on: t.string,
-    updated_by: t.union([t.undefined, t.string]),
-  },
-  'PolicyFile'
-);
+/**
+ * A group of items related to a data ingestion source (e.g. MySQL, nginx, AWS). Can include
+ * Kibana assets, ES assets, data source configuration templates, manual install steps, etc.
+ */
+export interface Package {
+  assets: Asset[];
+  description?: string;
+  name: string;
+  title?: string;
+  version: string;
+}
 
-export const NewRuntimeBackupPolicyFile = t.interface(
-  {
-    name: t.string,
-    description: t.string,
-    monitoring_enabled: t.boolean,
-    agent_version: t.string,
-    flat_data_sources: t.string,
-  },
-  'BackupPolicyFile'
-);
+/**
+ * Item installed for Kibana (e.g. dashboard, visualization), Elasticsearch (e.g. ingest
+ * pipeline, ILM policy), or a Kibana plugin (e.g. ML job)
+ */
+export interface Asset {
+  id: string;
+  type: AssetType;
+}
 
-const ExistingDocument = t.interface({
-  id: t.string,
-  updated_on: t.string,
-  updated_by: t.union([t.undefined, t.string]),
-});
+/**
+ * Types of assets which can be installed/removed
+ */
+export enum AssetType {
+  DataFrameTransform = 'data-frame-transform',
+  IlmPolicy = 'ilm-policy',
+  IndexTemplate = 'index-template',
+  IngestPipeline = 'ingest-pipeline',
+  MlJob = 'ml-job',
+  RollupJob = 'rollup-job',
+}
 
-export const RuntimeBackupPolicyFile = t.intersection([
-  NewRuntimeBackupPolicyFile,
-  ExistingDocument,
-]);
+/**
+ * A combination of an input type, the required config, an output, and any processors
+ */
+export interface Stream {
+  config?: { [key: string]: any };
+  id: string;
+  input: Input;
+  output_id: string;
```

Comment:
only used an output ID. Pulling in the output on its own for the agent as this is how the agent expects it after the most recent changes